wp-smartcrop/wp-smartcrop.php

```diff
@@ -736,14 +727,53 @@ private function get_id_and_size_from_tag( $tag ) {
 						break;
 					}
 				}
-				return $ret_val;
+				if ( $ret_val[0] && $ret_val[1] ) {
+					return $ret_val;
+				}
 			}
-			return false;
+			if ( $tag['attributes']['src'] ?? false && ( ! $ret_val[0] || ! $ret_val[1] ) ) { // try to get data from src
+				if ( ! $ret_val[0] ) { // missing id
+					$ret_val[0] = attachment_url_to_postid( $tag['attributes']['src'] );
+					if ( ! $ret_val[0] ) { // our last attempt. Try to derive original image url
+						$url_pattern = '/(.*)-[0-9]*x[0-9]*(\..*)/';
```

Comment:
Might be helpful to include some example matches for this regex in a code comment or a little more explanation inline of why we have to regex it. I know a lot of people find regexes hard to read.

Reply:
Added additional commenting here and for line 753 below.